core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -953,6 +953,9 @@ private Object readResolve() {
     @Nonnull
     @Restricted(NoExternalUse.class)
     public InstallState getInstallState() {
+        if (installState == null || installState.name() == null) {
```

Comment:
This approach is hackish. It won't be extensible if you add new fields to the InstallState